charmcraft/models/config.py

```diff
@@ -58,8 +58,19 @@ class JujuBooleanOption(_BaseJujuOption, frozen=True):
     default: bool | None = None
 
 
+class JujuSecretOption(_BaseJujuOption, frozen=True):
+    """A Juju option field containing a secret ID."""
+
+    type: Literal["secret"]
+    # A secret doesn't really make sense, since it's unlikely
+    # that anyone would know what the secret ID (specific to
+    # the deployment in a model) is at the time that they are
+    # writing the config, but included for completeness.
```

Comment:
Thanks for this comment! I presume Juju does support setting the default like this? (If not you can just set it to `None`)

Reply:
It does. Maybe there's some use-case that I'm not seeing but I imagine Juju supports it just because it's handling default values consistently across types, rather than because it's of any use here. I guess you could use it in dev maybe?

Reply:
Awesome, thanks!